synapseclient/core/version_check.py

```diff
@@ -15,84 +15,176 @@
 import json
 import re
 import sys
+import urllib.request
+from typing import Optional
 
 import requests
 
 import synapseclient
 
 _VERSION_URL = "https://raw.githubusercontent.com/Sage-Bionetworks/synapsePythonClient/master/synapseclient/synapsePythonClient"  # noqa
+_PYPI_JSON_URL = "https://pypi.org/pypi/synapseclient/json"
+_RELEASE_NOTES_URL = "https://python-docs.synapse.org/news/"
 
 
 def version_check(
-    current_version=None, version_url=_VERSION_URL, check_for_point_releases=False
-):
+    current_version: Optional[str] = None,
+    check_for_point_releases: bool = False,
+    use_local_metadata: bool = False,
+) -> bool:
     """
     Gets the latest version information from version_url and check against the current version.
     Recommends upgrade, if a newer version exists.
 
-    Arguments:
-        current_version: The current version of the entity
-        version_url: The URL of the entity version
-        check_for_point_releases: Bool.
+    This wraps the _version_check function in a try except block.
+    The purpose of this is so that no exception caught running the version check stops the client from running.
+
+    Args:
+        current_version (Optional[str], optional): The current version of the package.
+          Defaults to None.
+          This argument is mainly used for testing.
+        check_for_point_releases (bool, optional):
+          Defaults to False.
+          If True, The whole package versions will be compared (ie. 1.0.0)
+          If False, only the major and minor package version will be compared (ie. 1.0)
+        use_local_metadata (bool, optional):
+          Defaults to False.
+          If True, importlib.resources will be used to get the latest version fo the package
```

Comment:
nit: The typing in the docstring is not needed. The type is implied based on the typing in the function signature.